odl/solvers/functional/default_functionals.py

```diff
@@ -0,0 +1,731 @@
+
+        if constant not in self.range:
+            raise TypeError('constant {} not in the range {}.'
+                            ''.format(constant, self.range))
+
+        self._constant = constant
+
+    @property
+    def constant(self):
+        return self._constant
+
+    def _call(self, x):
+        """Applies the functional to the given point.
+
+        Parameters
+        ----------
+        x : `LinearSpaceVector`
+            Element in the domain of the functional.
+
+        Returns
+        -------
+        out : `field` element
+            Result of the evaluation.
+        """
+        return self._constant
+
+    @property
+    def gradient(self):
+        """Gradient operator of the functional.
+
+        Notes
+        -----
+        The operator that corresponds to the mapping
+
+        .. math::
+
+            x \\to \\nabla f(x)
+
+        where :math:`\\nabla f(x)` is the element used to evaluate
+        derivatives in a direction :math:`d` by
+        :math:`\\langle \\nabla f(x), d \\rangle`.
+        """
+        return ZeroOperator(self.domain)
+
+    def proximal(self, sigma=1.0):
+        """Return the proximal operator of the functional.
+
+        The proximal operator for the constant functional is the identity
+        operator.
+
+        Parameters
+        ----------
+        sigma : positive float, optional
+            Step size-like parameter
+
+        Returns
+        -------
+        out : Operator
+            Domain and range equal to domain of functional
+        """
+
+        return IdentityOperator(self.domain)
+
+    @property
+    def conjugate_functional(self):
+        """Convex conjugate functional of the constant functional.
+
+        This functional is defined as
+
+            ``f^*(x) = -constant if x=0, infty else``.
+        """
+        functional = self
+
+        class ConjugateToConstantFunctional(Functional):
+
+            """The convex conjugate functional of this functional."""
+
+            def __init__(self):
+                """Initialize a new instance."""
+                super().__init__(functional.domain, linear=False)
+                self._zero_element = self.domain.zero()
+                self._constant = functional._constant
+
+            @property
+            def constant(self):
+                """Returns the constant that the constant functional maps
+                to.
+                """
+                return self._constant
+
+            def _call(self, x):
+                """Applies the functional to the given point.
+
+                Parameters
+                ----------
+                x : `LinearSpaceVector`
+                    Element in the domain of the functional.
+
+                Returns
+                -------
+                out : `field` element
+                    Result of the evaluation.
+                """
+
+                if x == self._zero_element:
```

Comment:
```
x == self._zero_element:
```

will at some point call `norm` anyway, so this is not faster, only more memory  hungry.